sway-lib-std/src/storage.sw

```diff
@@ -681,6 +681,262 @@ impl<V> StorageVec<V> {
     pub fn clear(self) {
         let _ = clear::<u64>(__get_storage_key());
     }
+
+    /// Swaps two elements.
+    ///
+    /// ### Arguments
+    ///
+    /// * element1_index - The index of the first element.
+    /// * element2_index - The index of the second element.
+    ///
+    /// ### Reverts
+    ///
+    /// * If `element1_index` or `element2_index` is greater than the length of the vector.
+    ///
+    /// ### Storage Access
+    ///
+    /// * Reads: `3`
+    /// * Writes: `2`
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// use std::storage::StorageVec;
+    ///
+    /// storage {
+    ///     vec: StorageVec<u64> = StorageVec {}
+    /// }
+    ///
+    /// fn foo() {
+    ///     storage.vec.push(5);
+    ///     storage.vec.push(10);
+    ///     storage.vec.push(15);
+    ///
+    ///     storage.vec.swap(0, 2);
+    ///     assert(15 == storage.vec.get(0).unwrap());
+    ///     assert(10 == storage.vec.get(1).unwrap());
+    ///     assert(5 == storage.vec.get(2).unwrap());
+    /// ```
+    #[storage(read, write)]
+    pub fn swap(self, element1_index: u64, element2_index: u64) {
+        let len = get::<u64>(__get_storage_key()).unwrap();
+        assert(element1_index < len);
+        assert(element2_index < len);
+
+        if element1_index == element2_index {
+            return;
+        }
+
+        let element1_key: b256 = sha256((element1_index, __get_storage_key()));
+        let element2_key: b256 = sha256((element2_index, __get_storage_key()));
```

Comment:
```suggestion
        let element1_key = sha256((element1_index, __get_storage_key()));
        let element2_key = sha256((element2_index, __get_storage_key()));
```

Does the compiler force these type annotations? I'm not convinced that they are beneficial here.